backend/siarnaq/api/episodes/challonge.py

```diff
@@ -68,31 +74,149 @@ def bulk_add_participants(tournament_url, participants):
+                round_indexes.add(round_index)
+
+    return round_indexes
+
+
+def get_match_and_participant_objects_for_round(tournament_id, round):
+    tournament_data = get_tournament_data(tournament_id)
+    # Derive match dicts/JSON objects (that Challonge gives us) of this round
+    challonge_matches = []
+    # Also derive participant dicts/JSON objects that Challonge gives us,
+    # and map them with IDs for easy lookup
+    challonge_participants = dict()
+
+    for item in tournament_data["included"]:
+        match item:
+            case {
+                "type": "match",
+                "attributes": {"round": round.challonge_id, "state": state},
+            }:
+                # Only enqueue the round if all matches are "open".
+                # NOTE: it would be good to have a "force re-enqueue round",
+                # which re-enqueues matches even if matches or round
+                # already in progress.
+                # This would change the following check --
+                # matches could be open _or done_.
+                # !!! This is also _really hard_ right now
+                # cuz it involves match deletion which is really hard.
+                # Track in #549
+                if state != "open":
+                    # For later, have this raise a more specific exception.
+                    # Then have the caller handle this return
+                    # and translate it into an HTTP response.
+                    raise RuntimeError(
+                        "The bracket service's round does not only\
+                            have matches that are ready."
+                    )
+                challonge_matches.append(item)
+
+            case {"type": "participant", "id": id}:
+                challonge_participants[id] = item
+
+    match_objects = []
+    match_participant_objects = []
+
+    for challonge_match in challonge_matches:
+        match_object = apps.get_model("compete", "Match")(
+            episode=round.tournament.episode,
+            tournament_round=round,
+            alternate_order=True,
+            is_ranked=False,
+            challonge_id=challonge_match["id"],
+        )
+        match_objects.append(match_object)
+
+        # Note that Challonge 1-indexes its player indexes
+        # while Saturn 0-indexes.
+        for (challonge_player_index, saturn_player_index) in (
+            ("player1", 0),
+            ("player2", 1),
+        ):
+            # This looks ugly but it's how to parse through the Challonge-related data.
+            challonge_participant_id = challonge_match["relationships"][
+                challonge_player_index
+            ]["data"]["id"]
+            challonge_participant_misc_key = challonge_participants[
+                challonge_participant_id
+            ]["attributes"]["misc"]
+            team_id, submission_id = (
+                int(_) for _ in challonge_participant_misc_key.split(",")
```

Comment:
I assume you're in full control of what's in the misc key. Can this be json / something with more structure than a tuple?